/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// ms_service.proto
syntax = "proto3";

package ms_serving;

service MSService {
  rpc Predict(PredictRequest) returns (PredictReply) {}
  rpc Test(PredictRequest) returns (PredictReply) {}
}

message PredictRequest {
  repeated Tensor data = 1;
  repeated Images images = 2;
}

message PredictReply {
  repeated Tensor result = 1;
}

enum DataType {
  MS_UNKNOWN = 0;
  MS_BOOL = 1;
  MS_INT8 = 2;
  MS_UINT8 = 3;
  MS_INT16 = 4;
  MS_UINT16 = 5;
  MS_INT32 = 6;
  MS_UINT32 = 7;
  MS_INT64 = 8;
  MS_UINT64 = 9;
  MS_FLOAT16 = 10;
  MS_FLOAT32 = 11;
  MS_FLOAT64 = 12;
}

message TensorShape {
  repeated int64 dims = 1;
};

message Tensor {
  // tensor shape info
  TensorShape tensor_shape = 1;

  // tensor content data type
  DataType tensor_type = 2;

  // tensor data
  bytes data = 3;
}

message Images{
  repeated bytes images = 1;
  uint32 input_index = 2;
}
